### PR TITLE
PG: Updates to config

### PR DESCRIPTION
**Pull request by jamessewell**
- added some values in postgresql.conf
- added support for sync replication
- changed creation of REPLICATION user to be async to avoid deadlock
- added an option to use data checksums for initdb

### CODE REVIEW

File: postgresql/default.toml [hunk 1]
```diff
@@ -11,6 +12,8 @@ password = 'admin'
 [replication]
 name = 'replication'
 password = 'replication'
+synchronous = true
```
Comment (bodymindarts) [inline postgresql/default.toml]: Don't think this should be on by default

File: postgresql/default.toml [hunk 2]
```diff
@@ -11,6 +12,8 @@ password = 'admin'
...
+synchronous_standby_names = '*'
```
Comment (bodymindarts) [inline postgresql/default.toml]: Don't know if this should be editable. Ideally hab would just figure everything out by itself.

File: postgresql/hooks/init [hunk 3]
```diff
@@ -20,7 +20,7 @@ if [[ ! -f "{{pkg.svc_data_path}}/PG_VERSION" ]]; then
     -E utf8 \
     -D {{pkg.svc_data_path}} \
     --pwfile {{pkg.svc_config_path}}/pwfile \
-    --locale POSIX
+    --locale POSIX {{ #if cfg.data_checksums }}--data-checksums{{ /if }}
```
Comment (bodymindarts) [inline postgresql/hooks/init]: From reading a bit about this setting I would say we have it on by default or? It protects against data corruption but doesn't seem to have any downside to it.

File: postgresql/default.toml [hunk 4]
```diff
@@ -1,7 +1,8 @@
 port = '5432'
 max_connections = 500
-log_line_prefix = "%m: "
+log_line_prefix = '%t [%p]: [%l-1] user=%u,db=%d,client=%h %r (%x:%e)'
 log_level = 'ERROR'
+data_checksums = true
```
Comment (bodymindarts) [inline postgresql/default.toml]: Why would anyone wan't to turn this off? Does it really need to be configurable?

**Review by bodymindarts — CHANGES_REQUESTED** [review]

File: postgresql/default.toml (hunk 4 above)
Comment (bodymindarts) [reply postgresql/default.toml]: Okay so it makes things a bit slower.... so it allows ppl to choose fast over correct I guess.

Comment (jamessewell) [reply postgresql/default.toml]: Good point actually, I think you are right and we just leave it on till there is a use case for off?

File: postgresql/default.toml (hunk 1 above)
Comment (jamessewell) [reply postgresql/default.toml]: Agree

File: postgresql/default.toml (hunk 2 above)
Comment (jamessewell) [reply postgresql/default.toml]: I think we have a lot of customers who would want this - but to make it useful you actually need to manage the `application_name` of the standby servers. Probably too complex for the initial release. 

I'll take it out and always use * and maybe return to it later.

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**